### PR TITLE
REPO-2184: Community: ALF-21875: NPE in TagScopePropertyMethodInterceptor due to Script API conversion of null ContentData

### PR DESCRIPTION
**Pull request by anechifor**
ALF-21875: NPE in TagScopePropertyMethodInterceptor due to Script API conversion of null ContentData

### CODE REVIEW

File: src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java [hunk 1]
```diff
@@ -163,6 +171,7 @@ public void setUp() throws Exception
         this.checkOutCheckInService = (CheckOutCheckInService) ctx.getBean("CheckoutCheckinService");
         this.actionService = (ActionService)ctx.getBean("ActionService");
         this.transactionService = (TransactionService)ctx.getBean("transactionComponent");
+        this.siteService       = ctx.getBean("SiteService", SiteService.class);
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: Please remove extra tabs/white lines.

File: src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java [hunk 2]
```diff
@@ -2426,5 +2435,89 @@ public Void execute() throws Throwable
...
+
+                        NodeRef containerTagScope = fileFolderService
+                                        .create(documentLibraryOfSite, "containerTagScope" + GUID.generate(), ContentModel.TYPE_FOLDER)
+                                        .getNodeRef();
+                        
+        //STEP4 create tag
+        String tagName = GUID.generate();
+                taggingService.createTag(storeRef, tagName);
+                        
+                        // Add some tag scopes
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: Something is wrong with the formatting here, tabs perhaps.

File: src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java [hunk 3]
```diff
@@ -2426,5 +2435,89 @@ public Void execute() throws Throwable
...
+        // STEP6 execute script
+        // Create a model to pass to the unit test scripts
+        Map<String, Object> model = new HashMap<String, Object>();
+        model.put("customSiteName", siteName);
+
+        // Execute the unit test script
+        ScriptLocation location = new ClasspathScriptLocation("org/alfresco/repo/site/script/test_tagScopeALF_21875.js");
+        this.scriptService.executeScript(location, model);
+        
+         TagScopePropertyMethodInterceptor.setEnabled(Boolean.FALSE);
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: Please, remove unnecessary space and whiteline.

File: src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java [hunk 4]
```diff
@@ -2426,5 +2435,89 @@ public Void execute() throws Throwable
             nodeRefPropInterceptor.setFilterOnGet(true);
         }
     }
+    
+    /**
+     * ALF-21875
+     */
+    @Test
+    public void testTagScopeALF_21875() {
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: Please rename the test to meaningful name and _not_ include the JIRA numbers.

Comment (killerboot) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: The Brace should start from new line, see the [coding standarts](https://community.alfresco.com/docs/DOC-4658-coding-standards)

File: src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java [hunk 5]
```diff
@@ -2426,5 +2435,89 @@ public Void execute() throws Throwable
...
+
+                        // STEP2 upload a document in documentLibrary
+                        NodeRef documentLibraryOfSite = siteService.getContainer(siteInfo.getShortName(), "doclib");
+
+                        NodeRef containerTagScope = fileFolderService
+                                        .create(documentLibraryOfSite, "containerTagScope" + GUID.generate(), ContentModel.TYPE_FOLDER)
+                                        .getNodeRef();
+                        
+        //STEP4 create tag
+        String tagName = GUID.generate();
```
Comment (mmuller88) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: Not sure if the rendering in Github is wrong but that looks not in line and I can not see STEP3

File: src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java [hunk 6]
```diff
@@ -2426,5 +2435,89 @@ public Void execute() throws Throwable
...
+     * ALF-21875
+     */
+    @Test
+    public void testTagScopeALF_21875() {
+        
+        TagScopePropertyMethodInterceptor.setEnabled(Boolean.TRUE);
+        AuthenticationUtil.setFullyAuthenticatedUser(AuthenticationUtil.getAdminUserName());
+        String siteName = GUID.generate();
+
+        this.transactionService.getRetryingTransactionHelper().doInTransaction(new RetryingTransactionCallback<Void>() {
```
Comment (mmuller88) [inline src/test/java/org/alfresco/repo/tagging/TaggingServiceImplTest.java]: Please keep Alfresco syntax formation we do { in the next line. For the others as well.

**Comment by anechifor** [comment]
REPO-2184: Community: ALF-21875: NPE in TagScopePropertyMethodInterceptor due to Script API conversion of null ContentData

**Pull request merged**